docs/architecture/adr-001-redesign-knuu-async-approach.md

```diff
@@ -0,0 +1,49 @@
+# ADR 001: Redesigning Knuu for Asynchronous Testing Approach
+
+## Status
+
+Draft
+
+## Context
+
+The existing design of `Knuu` requires exposing the addresses and ports of the target instance running alongside BitTwister, complicating the configuration process. This setup requires Knuu to orchestrate the instances, configure them, and execute the test, leading to complexities in managing traffic shaping. Moreover, as the current model relies on port forwarding, scalability issues have surfaced, causing inconsistent errors and hindering seamless test execution.
+
+## Problem
+
+The current synchronous model necessitates exposing the target instance's address and ports to BitTwister via a sidecar. This centralized approach introduces dependencies and complexities in configuring the test environment. Additionally, the reliance on a proxy approach or port forwarding poses scalability issues, resulting in inconsistent errors during test execution.
+
+## Proposed Solution
+
+### A Distributed Approach with a Playbook Model
+
+Embrace a distributed playbook model similar to Testground, allowing asynchronous orchestration and execution of test scenarios:
+
+1. **Test Scenario Definition:** Users define test scenarios in a descriptive language (e.g., Golang).
+2. **Knuu Instance Initiation:** Knuu initiates instances and assigns specific playbooks to each instance.
+3. **BitTwister Sidecar with Playbook Understanding:** BitTwister runs alongside instances with added functionality to interpret and execute the playbook's instructions.
+4. **Dynamic Traffic Shaping:** BitTwister adjusts traffic shaping based on predefined events or triggers from the target instance, eliminating the need for continuous centralized control.
+
+### Advantages
+
+1. **Decentralized Execution:** Instances operate autonomously based on pre-defined playbooks, reducing dependency on centralized control.
+2. **Elimination of IP Exposure:** Removes the necessity to expose IP addresses or utilize proxies for controlling data.
```

Comment:
Private IPs is the king of the game here. I would add that all test containers need only private IPs to talk to one another in a cluster. 

When we do multiclusters - we will figure that out -> this point should be pointed in Considerations / Risks point